### DISCUSSION TITLE
Deno LSP activating inlayHints

### DISCUSSION THREAD

**Discussion by gvelim** [zed-industries/zed#15620] [Feature Requests]
I have enabled Deno LSP on my settings which works however without inlayHints. I have [looked around for any settings](https://github.com/zed-industries/zed/issues/11366#issuecomment-2093471927) to enable inlayHints but with no luck.

Does anyone know if inlayHints are supported and if yes point to the relevant settings ?

PS: Deno extension version v0.0.2; Zed 0.146.3

```
    // LSP Specific settings.
    "lsp": {
        "deno": {
              "settings": {
                "deno": {
                  "enable": true
                }
            }
        }
    },
```

**Comment by eveffer** [comment]
After many attempts, I've finally gotten the inlay hints working with deno lsp. This #11366 Issue is what provided the info that helped me resolve this. 

What was confusing was that the deno lsp was activating because there was a deno.json file in the workspace which was overriding anything set in the zed settings.json so I couldn't tell what was working and what wasn't.

Additionally, deno lsp will NOT send back inlay hint information unless it's explicitly enabled via the `inlayHints` properties in the deno specific settings that are sent to the lsp for the workspace, so even though I could see in the language server RPC messages that the editor was sending a request for inlay hints, the deno lsp was responding with `null`.

This is what turned out to be the syntax for enabling inlay hints with deno.

```json
{
  "lsp": {
    "deno": {
      "settings": {
        "deno": {
          "enable": true,
          "inlayHints": {
            "parameterNames": {
              "enabled": "all"
            },
            "parameterTypes": {
              "enabled": true
            },
            "variableTypes": {
              "enabled": true
            },
            "propertyDeclarationTypes": {
              "enabled": true
            },
            "functionLikeReturnTypes": {
              "enabled": true
            },
            "enumMemberValues": {
              "enabled": true
            }
          }
        }
      }
    }
  },

  "languages": {
    "TypeScript": {
      "language_servers": ["deno"],
      "inlay_hints": {
        "enabled": true,
        "show_type_hints": true,
        "show_parameter_hints": true,
        "show_other_hints": true
      },
      "formatter": "language_server"
    }
  }
}
```

**Reply by gvelim** [reply]
many thanks! inlay hints now work

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
